svea_vision/scripts/pedestrian_flow_estimate.py

```diff
@@ -0,0 +1,229 @@
+            
+                if len(self.vy_smoothed_dict[person_id]) >= 2:
+                    ay = (self.vy_smoothed_dict[person_id][-1]-self.vy_smoothed_dict[person_id][-2])/dt
+                    ax = (self.vx_smoothed_dict[person_id][-1]-self.vx_smoothed_dict[person_id][-2])/dt
+                    if person_id in self.ay_dict:
+                        self.ay_dict[person_id].append(ay)
+                        self.ax_dict[person_id].append(ax)
+                    else:
+                        self.ay_dict[person_id] = deque([ay], maxlen=self.SPEED_ACCELERATION_LENGTH)
+                        self.ax_dict[person_id] = deque([ax], maxlen=self.SPEED_ACCELERATION_LENGTH)
+
+                    if len(self.ay_dict[person_id]) >= self.FREQUENCY_ACC:
+                        smoothed_ay = self.low_pass_filter(self.ay_dict[person_id],self.FREQUENCY_ACC)
+                        smoothed_ax = self.low_pass_filter(self.ax_dict[person_id],self.FREQUENCY_ACC)
+
+        return smoothed_vx, smoothed_vy, smoothed_ax, smoothed_ay
+
+
+    def inaccurate_position_estimate(self, person_id, current_x, current_y, current_time):
+        """
+        This function detects substantially unpredicted, hence wrong, position jumps and discard the person_id 
+        from the deques if that happens. (Tested on real data)
+        Notice that this check is performed only when at least one smoothened velocity has been computed, so after some time since the person 
+        is firstly detected.
+        """
+        dt = current_time - self.time_dict[person_id][-1] 
+        predicted_x = self.x_dict[person_id][-1] + self.vx_smoothed_dict[person_id][-1] * dt
+        predicted_y = self.y_dict[person_id][-1] + self.vy_smoothed_dict[person_id][-1] * dt
+
+        if abs(current_x - predicted_x) > self.DISCARD_THRESHOLD  or abs(current_y - predicted_y) > self.DISCARD_THRESHOLD :
+            return True
+        else:
+            return False
+
+    def __clean_up_dict(self, current_time):
+        ids_to_drop = []
+        for id, value in self.time_dict.items():
+            last_time = value[-1]
+            if current_time - last_time >= self.MAX_TIME_MISSING:
+                ids_to_drop.append(id)
+        self.__drop_ID(ids_to_drop)
+
+    def __drop_ID(self,ids_to_drop):
+        for id in ids_to_drop:
+            # remove id deque from dictonaries if present, otherwise return None.
+            self.person_states.pop(id, None)
+            self.x_dict.pop(id, None)
+            self.y_dict.pop(id, None)
+            self.time_dict.pop(id, None)
+            self.vy_dict.pop(id, None)
+            self.vx_dict.pop(id, None)
+            self.vy_smoothed_dict.pop(id, None)
+            self.vx_smoothed_dict.pop(id, None)
+            self.ay_dict.pop(id, None)
+            self.ax_dict.pop(id, None)
+
+    def start(self):
+        self.__listener()
+
+
+if __name__ == "__main__":
+    predictions = PedestrianFlowEstimator()
```

Comment:
Minor note:

I prefer the start to be called from here and not in `__init__`. It should only be initialization there! You also do not need to assign it,

```suggestion
  PedestrianFlowEstimator().start()
```